tests/integration/src/fluvio-test-macro/src/lib.rs

```diff
@@ -0,0 +1,114 @@
+
+    let out_fn_iden = Ident::new(&test_name.to_string(), Span::call_site());
+
+    // TODO: Do we want this to be given via attributes, or randomized?
+    let namespace: String = "default".to_string(); // May be UUID or pseudo-random unique english name
+
+    let output_fn = quote! {
+
+        #[test] #[ignore]
+        fn #out_fn_iden() {
+            use fluvio_future::task::run_block_on;
+            use fluvio_future::timer::sleep;
+            use fluvio_cluster::{ClusterConfig, ClusterConfigBuilder, ClusterError, ClusterInstaller};
+            use fluvio::{Fluvio, FluvioConfig};
+            use fluvio_cluster::ClusterUninstaller;
+            use std::time::Duration;
+            use fluvio_test_options::{TestOption, ProductOption};
+            use fluvio_system_util::bin::get_fluvio;
+            use serde::Deserialize;
+
+            fluvio_future::subscriber::init_logger();
+
+            // Magic variable
+            let test_option : TestOption = serde_json::from_str(#test_opt_str).unwrap();
+
+            run_block_on(async move {
+                fn cleanup_cluster() {
+                    run_block_on(async move {
+                        //sleep(Duration::from_secs(30)).await;
+
+                        println!("Deleting cluster");
+
+                        // Delete cluster
+                        let uninstaller = ClusterUninstaller::new()
+                            .with_namespace(#namespace)
+                            .build().unwrap();
+                        uninstaller.uninstall().await.expect("");
+                    });
+                }
+
+
+                // Verify that we can reach a fluvio exe before starting test
+                let _ = get_fluvio().expect("no fluvio");
+
+
+                // TODO: We want the option to skip clean up if the test fails
+                std::panic::set_hook(Box::new(|panic_info| {
+                    println!("{:?}", &panic_info);
+                    cleanup_cluster();
+                }));
+
+
+                // TODO: SPU replicas need to be parsed and available while building the cluster config
+
+
+                println!("Starting cluster");
+
+                // Initialize kubernetes
+                let namespace: String = #namespace.to_string();
+                let cluster_config: ClusterConfig = fluvio_cluster::ClusterConfig::builder("0.7.0-alpha.5")
+                    .namespace(&namespace)
+                    //.spu_replicas(self.option.spu)
+                    .skip_checks(true)
+                    .save_profile(true)
+                    .build()
+                    .unwrap();
+                let installer = ClusterInstaller::from_config(cluster_config).unwrap();
+                let status = installer.install_fluvio().await.expect("");
+
+                // Magic variables
+                let fluvio_config = FluvioConfig::new(status.address());
+                let fluvio = Fluvio::connect_with_config(&fluvio_config).await.unwrap();
```

Comment:
Since `cleanup_cluster` is its own function, it may make sense to put this in a function like `setup_cluster`. Also note that this code is sitting directly next to the test body, so all of these values would become accessible via "magic values" in the body, which I don't think is great